src/main/java/com/atherys/towns/service/TownService.java

```diff
@@ -336,6 +336,12 @@ public void decreaseTownSize(Town town, int amount) {
         townRepository.saveOne(town);
     }
 
+    public Set<Player> getTownOnlinePlayers(Town town) {
+        Set<UUID> townResidents = town.getResidents().stream().map(Resident::getId).collect(Collectors.toSet());
+        return Sponge.getServer().getOnlinePlayers().stream().filter(
+                player -> townResidents.contains(player.getUniqueId())).collect(Collectors.toSet());
+    }
+
```

Comment:
Same here, there is already a method for this in TownFacade, although I like yours better since it doesn't have to call so many functions. Same suggestion for moving it to the facade.

Reply:
Moved to the TownFacade function, but replaced the implementation